src/wallet/wallet.cpp

```diff
@@ -174,39 +174,40 @@ std::shared_ptr<CWallet> CreateWallet(interfaces::Chain& chain, const std::strin
     WalletLocation location(name);
     if (location.Exists()) {
         error = "Wallet " + location.GetName() + " already exists.";
-        status = WalletCreationStatus::CREATION_FAILED;
-        return nullptr;
+        return WalletCreationStatus::CREATION_FAILED;
     }
 
     // Wallet::Verify will check if we're trying to create a wallet with a duplicate name.
     std::string wallet_error;
     if (!CWallet::Verify(chain, location, false, wallet_error, warning)) {
         error = "Wallet file verification failed: " + wallet_error;
-        status = WalletCreationStatus::CREATION_FAILED;
-        return nullptr;
+        return WalletCreationStatus::CREATION_FAILED;
+    }
+
+    // Do not allow a passphrase when private keys are disabled
+    if (!passphrase.empty() && (wallet_creation_flags & WALLET_FLAG_DISABLE_PRIVATE_KEYS)) {
+        error = "Passphrase provided but private keys are disabled.";
+        return WalletCreationStatus::CREATION_FAILED;
     }
 
     // Make the wallet
-    std::shared_ptr<CWallet> wallet = CWallet::CreateWalletFromFile(chain, location, wallet_creation_flags);
+    wallet = CWallet::CreateWalletFromFile(chain, location, wallet_creation_flags);
```

Comment:
Should not set wallet yet otherwise it leaks to the caller if something below fails. My suggestion is to rename the argument to `std::shared_ptr<CWallet>& result)` and below do:
```cpp
    result = wallet;
    return WalletCreationStatus::SUCCESS;
}
```

Reply:
I added this, thanks!

Reply:
Looks good. I don't love the name `return` for the wallet argument (since it's a pointer that's passed in and then updated), but I'm not going to bikeshed this PR over argument naming.